### PR TITLE
Frontend Multistage Docker Build

### LINKED ISSUE #923: Core-v4 Frontend Multistage Docker Build
This involves changing a single Dockerfile for SCE’s frontend. It requires learning the basics of docker.

When we build Core-v4's frontend for production. The size of the container is 1.5GB. While @imdominicreed helped reduce the size and build time with #916, we can take it even further. This involves using multistage builds, where we run `npm run build` in a base image and then move only the minified files into a secondary image to be served. By completing this,
the Core-v4 frontend file could be brought down to a size of about 300-400MB.
 
### Helpful links
- blog post on multistage react builds [link](https://medium.com/geekculture/dockerizing-a-react-application-with-multi-stage-docker-build-4a5c6ca68166)
- example Dockerfile for doing this
([link](https://github.com/evanugarte/LAN-File-Transfer/blob/master/website/Dockerfile))
 
### Image Sizes for Reference
```
Nginx:       133MB
Node Alpine: 117MB
Python Slim: 156MB
MongoDB:     701MB
```

### PR DESCRIPTION
**Pull request by crowbardispensingcore**
Resolve #923 
Reduce frontend docker image size down to 81.1MB from 1.5GB, improved by 95%.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ services:
     ports:
       - '3000:3000'
     restart: 'on-failure'
-    command: 'npm run deploy-production-frontend'
+    # command: 'npm run deploy-production-frontend'
```
Comment (crowbardispensingcore) [inline docker-compose.yml:63]:
```suggestion
```

File: docker/Dockerfile-frontend [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+FROM nginx:alpine
+
+COPY ./docker/nginx.conf /etc/nginx/conf.d/default.conf
+COPY --from=builder /frontend/build /usr/share/nginx/html
 
 EXPOSE 3000
 
-CMD npm run deploy-production-frontend
\ No newline at end of file
+CMD ["nginx", "-g", "daemon off;"]
```
Comment (crowbardispensingcore) [inline docker/Dockerfile-frontend:24]:
```suggestion
CMD ["nginx", "-g", "daemon off;"]

```

File: src/APIFunctions/User.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ export async function getAllUsers(token) {
   let status = new UserApiResponse();
   await axios
     // get all user!
-    .post(GENERAL_API_URL+'/User/users', {
+    .post(GENERAL_API_URL+'/User/users', {
```
Comment (crowbardispensingcore) [inline src/APIFunctions/User.js:19]:
```suggestion
    .post(GENERAL_API_URL+'/User/users', {
```

File: docker/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,10 @@
+server {
+  listen 3000;
+
+  location / {
+    root /usr/share/nginx/html;
+    index index.html;
+    try_files $uri $uri/ /index.html =404;
+  }
+  include /etc/nginx/extra-conf.d/*.conf;
+}
```
Comment (crowbardispensingcore) [inline docker/nginx.conf:10]:
```suggestion
}

```

**Comment by evanugarte** [comment]
1.5 GB = 1500 MB

81.1 MB / 1500 MB
image

insane!!!!

**Review by evanugarte — APPROVED** [review]
Tested and it works

image

**Review by imdominicreed — APPROVED** [review]
LGTM LGTM LGTM LGTM

**Pull request merged**